Cubical/Data/Int/Properties.agda

```diff
@@ -86,6 +98,288 @@ predSuc (pos n)          = refl
 predSuc (negsuc zero)    = refl
 predSuc (negsuc (suc n)) = refl
 
+sucDistMin : ∀ m n → sucℤ (min m n) ≡ min (sucℤ m) (sucℤ n)
```

Comment:
Wow, a lot of the proofs in this file are quite wild. Big pattern-matches where almost every case is `refl`.. Can one set things up so that this goes a bit smoother or is this necessary? Some tactics to solve the trivial cases automatically would be nice...

Reply:
I don't think I necessarily tried for all of these (sucDist and predDist namely), but I recall that attempting to consolidate the cases would run into issues. The problem is that min and max themselves are defined by pattern-matching, so they'll only actually compute once you apply the appropriate argument case.

Which means a lot of pattern matching for proofs about either.

Reply:
Hmm, I wonder if the best is to define max/min by deep pattern-matching like this? Wouldn't it be better to just use < or <= together with if-then-else? (Assuming that the comparison operations are bool valued...)

Reply:
Alternatively, define them with slightly less deep pattern-matching are use max/min for nat instead?

Reply:
I think part of the problem here is that they're defined by themselves rather than in terms of the equivalent operator on the nats. Given that I want to do a broader refactoring of Nat and Int (per our discussion), I'd rather that be its own PR instead of included in this one.

Reply:
Yeah, that makes sense. Please make a note of this and fix it in a future PR :-)